app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -51,15 +52,14 @@ import java.io.IOException
 import java.io.InputStream
 import java.io.OutputStream
 import java.net.InetAddress
-import java.util.ArrayList
 import javax.inject.Inject
 
 /**
  * Manager for the Wifi-P2p API, used in the local file transfer module
  */
 @SuppressWarnings("MissingPermission", "ProtectedMemberInFinalClass")
 class WifiDirectManager @Inject constructor(
-  private val activity: Activity,
```

Comment:
this doesn't need to be public and the 1 usage that demands it
```
  private suspend fun publishProgress(
    fileIndex: Int,
    fileName: String,
    fileStatus: FileItem.FileStatus
  ) {
    withContext(Dispatchers.Main) {
      wifiDirectManager.changeStatus(fileIndex, fileStatus)
      if (fileStatus == FileItem.FileStatus.ERROR) {
        val activity = wifiDirectManager.activity
        activity.toast(
          activity.getString(R.string.error_transferring, fileName),
          Toast.LENGTH_SHORT
        )
      }
    }
  }
```
is silly because it calls `changeStatus`:
```
  fun changeStatus(itemIndex: Int, status: FileStatus) {
    filesForTransfer[itemIndex].fileStatus = status
    callbacks?.onFileStatusChanged(itemIndex)
    if (status == FileStatus.ERROR) {
      activity.toast(
        activity.getString(
          R.string.error_transferring, filesForTransfer[itemIndex].fileName
        )
      )
    }
  }
```

Reply:
@macgills I don't quite  understand your example section
```java
 private suspend fun publishProgress(
    fileIndex: Int,
    fileName: String,
    fileStatus: FileItem.FileStatus
  ) {
    withContext(Dispatchers.Main) {
      wifiDirectManager.changeStatus(fileIndex, fileStatus)
      if (fileStatus == FileItem.FileStatus.ERROR) {
        val activity = wifiDirectManager.activity
        activity.toast(
          activity.getString(R.string.error_transferring, fileName),
          Toast.LENGTH_SHORT
        )
      }
    }
  }
```
In here we need an extra one file parameter of file name  because if file transfer fails then it will show an error message with the file name

Reply:
both show the filename
` activity.getString(R.string.error_transferring, fileName)`
` activity.getString(R.string.error_transferring, filesForTransfer[itemIndex].fileName)`

Reply:
D'oh! my eyes are glazed over too 😄